### PR TITLE
Release 3.0

### PR DESCRIPTION
**Pull request by denihs**
# Still in development
Current released version: ``3.0-rc.2``
To use it, please do a 
```bash
meteor update --release 3.0-rc.2
```
you can also use the following command to install meteor RC

```bash
npx meteor@rc
```
## Highlights
TODO
See more in the [milestones](https://github.com/meteor/meteor/milestone/101).

Preview docs: https://v3-docs.meteor.com/

## Maintainers checklist

⚠️⚠️⚠️ You can follow the updates on [What's left until an official Meteor 3.0? Discussion](https://github.com/meteor/meteor/discussions/12865)

### CODE REVIEW

File: packages/accounts-password/password_server.js
Comment (zarvox) [inline packages/accounts-password/password_server.js:539]: Can `Accounts.generateOptionsForEmail` also be made `async`, so that it can `await` each of the [user-provideable functions](https://github.com/meteor/meteor/blob/f60bfc7f75a24263539ef67fe856afab6eaf983e/packages/accounts-base/accounts_server.js#L1405-L1413) so that they too, may be async?

**Review by radekmie — COMMENTED** [review]
I reviewed the Minimongo and Mongo packages and added a few comments. Most of them have _many_ instances but I added only one comment each.

I also think that we should focus on addressing the comments from XXX first, as

File: packages/accounts-passwordless/passwordless_server.js
Comment (srsgores) [inline packages/accounts-passwordless/passwordless_server.js:192]:
```suggestion
      for (const { email, sequence } of tokens) {
    await Accounts.sendLoginTokenEmail({
      userId: user._id,
      sequence,
      email,
      ...(options.extra ? { extra: options.extra } : {}),
    });
  }
```

File: packages/ddp-client/test/stub_stream.js
Comment (srsgores) [inline packages/ddp-client/test/stub_stream.js:49]: Would this be clearer, or are we following a "node" convention here with `cb`?

```suggestion
    for (const callback of self.callbacks['reset']) {
```

File: packages/ddp-server/livedata_server_async_tests.js
Comment (srsgores) [inline packages/ddp-server/livedata_server_async_tests.js:25]: Remove these `console.log`s.

File: packages/meteor/dynamics_nodejs.js
Comment (srsgores) [inline packages/meteor/dynamics_nodejs.js:131]: Can this not be `const`?

```suggestion
    const runWithEnvironment = function() {
```

File: packages/meteor/dynamics_nodejs.js
Comment (srsgores) [inline packages/meteor/dynamics_nodejs.js:128]: Is `/* arguments */` here just to please `eslint`?

File: packages/meteor/dynamics_nodejs.js
Comment (srsgores) [inline packages/meteor/dynamics_nodejs.js:118]: Does this need to be `var`?

```suggestion
    const description = onException || 'callback of async function';
```

File: packages/meteor/dynamics_nodejs.js
Comment (srsgores) [inline packages/meteor/dynamics_nodejs.js:139]: Would this be clearer if we renamed these variables?

```suggestion
            resolvedAsyncValue = await callback.apply(_this, args);
```

**Review by srsgores — CHANGES_REQUESTED** [review]
Note that I'm not a maintainer and have just provided feedback from an outside perspective.  It's very possible that my suggestions violate the coding standards implemented for this project, so please feel free to disregard the feedback at your own discretion.

**Comment by Grubba27** [comment]
Current mocha runner for `alpha-9` is `grubba:mocha@3.0.1-alpha300.9`

File: packages/meteor/dynamics_nodejs.js
Comment (Torgen) [inline packages/meteor/dynamics_nodejs.js:55]: The original restored the saved value. This clobbers it with undefined.

File: packages/meteor/dynamics_nodejs.js
Comment (Torgen) [inline packages/meteor/dynamics_nodejs.js:62]: Does this mean I can't call withValue from inside another withValue call? If I set the same variable, the inner call will clobber the value with undefined, and if I set a different variable, I can't access the originally modified variable because the slot won't match.

Comment (Torgen) [reply packages/meteor/dynamics_nodejs.js:62]: I wrote a test case for this to confirm: https://github.com/Torgen/meteor/commit/57b314b1a224f41b989a37b3b480e8dcc1824c09

**Comment by zodern** [comment]
I have a couple concerns about how Meteor 3 now clones a git repo when running `meteor create` (https://github.com/meteor/meteor/pull/12697#issue-1779347625):

1. It is very important that `meteor create --release <old version>` continues to work reliably. We occasionally make changes to the skeletons that use new Meteor features, which was fine in the past when each release had their own copy of the skeletons. What is the plan to ensure this continues to work? Given our recent track record with these types of problems, I am very concerned that this will break and be rarely fixed.

2. I occasionally create Meteor apps while offline, either because there is an internet outage on a work day, or I am traveling or in a country with very unreliable internet. Meteor has always tried to ensure it works offline, as long as the packages it needs are already download. The new process for creating Meteor apps breaks this.

I'm also surprised that the `--prototype` option is removed from Meteor 3. The PR only explains that Meteor should be opinionated about this, but doesn't give the reason for the opinion that the option should be removed. I am curious about why it is removed, and what the plan is for the tutorials since they use it.

**Comment by zodern** [comment]
I have a couple concerns about how Meteor 3 now clones a git repo when running `meteor create` (https://github.com/meteor/meteor/pull/12697#issue-1779347625):

1. It is very important that `meteor create --release <old version>` continues to work reliably. We occasionally make changes to the skeletons that use new Meteor features, which was fine in the past when each release had their own copy of the skeletons. What is the plan to ensure this continues to work? Given our recent track record with these types of problems, I am very concerned that this will break and be rarely fixed.

2. I occasionally create Meteor apps while offline, either because there is an internet outage on a work day, or I am traveling or in a country with very unreliable internet. Meteor has always tried to ensure it works offline, as long as the packages it needs are already download. The new process for creating Meteor apps breaks this.

I'm also surprised that the `--prototype` option is removed from Meteor 3. The PR only explains that Meteor should be opinionated about this, but doesn't give the reason for the opinion that the option should be removed. I am curious about why it is removed, and what the plan is for the tutorials since they use it.

**Comment by Grubba27** [comment]
> I occasionally create Meteor apps while offline, either because there is an internet outage on a work day, or I am traveling or in a country with very unreliable internet. Meteor has always tried to ensure it works offline, as long as the packages it needs are already download. The new process for creating Meteor apps breaks this.

> I'm also surprised that the --prototype option is removed from Meteor 3. The PR only explains that Meteor should be opinionated about this, but doesn't give the reason for the opinion that the option should be removed. I am curious about why it is removed, and what the plan is for the tutorials since they use it.

I had the same feeling, so that is the reason why I still maintained the cached version of skeletons. If for some reason, the network, for example, was missing, you can still use those cached ones. 

The ``--prototype`` flag is still there as well but is only used for the ones that are cached, it will work as the same way as today. I will commit it back to `help.txt`.

https://github.com/meteor/meteor/blob/fd6f66386cef9131fb02a76761c77c2a28ac42b0/tools/cli/commands.js#L1030-L1036


Regarding the 1st I had not yet given much thought to it as my vision was that almost every developer when creating a new project tries to create it with the latest stable version of the framework that is used.

**Comment by zodern** [comment]
> Regarding the 1st I had not yet given much thought to it as my vision was that almost every developer when creating a new project tries to create it with the latest stable version of the framework that is used.

A couple reasons why developers might use an old release:
1. Some of us in the community maintain packages or tools that work with a wide range of Meteor versions. Sometimes someone reports an issue with a specific release, and we need to be able to create an app with that release.
2. Some tools for Meteor apps have test scripts that run `meteor create` for each release and makes sure the tools works. For example, my meteor-docker project does this to make sure it works with Meteor 1.2 - the latest release, and I know other projects do the same.
3. Checking old releases is very useful for core developers when looking into a performance regression or a bug
4. It allows people to use a specific Meteor version when creating a reproduction for a bug

File: packages/meteor/dynamics_nodejs.js
Comment (Torgen) [reply packages/meteor/dynamics_nodejs.js:62]: Filed https://github.com/meteor/meteor/issues/12702

**Comment by StorytellerCZ** [comment]
Getting the same issue on Ubuntu 23.04

**Comment by gunnartorfis** [comment]
I'm also getting this error (Cannot enlarge memory arrays). 
macOS Ventura 13.0, Node 18, npm 9.6.7.

**Comment by Grubba27** [comment]
Meteor 3.0-alpha.18 is launched and some of PRs that were merged:

- https://github.com/meteor/meteor/pull/12773 Made the webapp API cleaner and removing references to `connect ` lib.
- https://github.com/meteor/meteor/pull/12869 Cleaned and caught some API's in `passwordless` that needed to be async.
- https://github.com/meteor/meteor/pull/12877 Merge the devel branch onto 3.0.
- https://github.com/meteor/meteor/pull/12875 Ensures we do not kill the dev server when we make a mistake while coding.
- https://github.com/meteor/meteor/pull/12860 Allow some old build plugins to run in Meteor 3 (this one might help many people while updating their apps)

Thanks to the contributors in this release:
- @denihs 
- @zodern 
- @filipenevola 
- @Grubba27

**Comment by filipenevola** [comment]
@ Grubba27 @denihs Something went wrong in this release.

I'm getting errors about `mongo-decimal` in codeftw.dev project (logs below) and also in the zCloud process to build our Docker image ([link](https://github.com/zcloud-ws/zcloud-community-images/actions/runs/6872688043/job/18691541027)):

```
=> Started proxy.
=> Errors prevented startup:
   
   While selecting package versions:
   error: Conflict: Constraint mongo-decimal@0.1.4-alpha300.18 is not satisfied
   by mongo-decimal 0.1.3.
   Constraints on package "mongo-decimal":
   * mongo-decimal@0.1.4-alpha300.18 <- mongo 2.0.0-alpha300.18
   * mongo-decimal@0.1.4-alpha300.18 <- minimongo 2.0.0-alpha300.18 <-
   allow-deny 2.0.0-alpha300.18 <- mongo 2.0.0-alpha300.18
   * mongo-decimal@0.1.4-alpha300.18 <- minimongo 2.0.0-alpha300.18 <- mongo
   2.0.0-alpha300.18
```

Also, in codeftw I "forced" the update by doing it manually.

My meteor --version says alpha 18, but when I run my project, I get the following:

```
Track: METEOR;
        Version: 3.0-alpha.17
```

I've been updating every alpha for a long time, and this was the first time I had these issues. Maybe the release was generated with the wrong dev bundle?

**Comment by StorytellerCZ** [comment]
@filipenevola we had the mongo-decimal issue before. I think that package needs to be release manually for some reason.

**Comment by Grubba27** [comment]
mongo-decimal package was released!

**Comment by zodern** [comment]
I am getting a similar error when trying to publish a package with Meteor 3 that has a weak dependency on http:

```
While selecting package versions:
error: Conflict: Constraint http@3.0.0 is not satisfied by http 1.4.1.
Constraints on package "http":
* http@3.0.0 <- montiapm:agent 3.0.0-beta.1
* http@3.0.0 <- local-test:montiapm:agent 3.0.0-beta.1
```

**Comment by zodern** [comment]
I found the cause. The HTTP package was never updated after merging https://github.com/meteor/meteor/pull/12671.

**Comment by Grubba27** [comment]
Published http@3.0.0-alpha300.5 @zodern

File: packages/minimongo/constants.js
Comment (vincentcarpentier) [inline packages/minimongo/constants.js:149]: Why don't letting backward compatibility on those methods server side with a synchronous version ?

Comment (filipenevola) [reply packages/minimongo/constants.js:149]: It's not possible to use these methods without async without Fibers (aka Meteor 3). You would get back a not resolved Promise.

**Comment by xet7** [comment]
@Grubba27 

Please revert or fix that "Update-source-map-support"  https://github.com/meteor/meteor/pull/12359/commits/dc73543c8a15b4956596c12b2aaf78382351afbe , because it causes this error:

```
$ meteor create --blaze armtest
modules: updating npm dependencies -- @meteorjs/reify, meteor-babel-helpers...
/home/wekan/.meteor/tools/tool-env/source-map-retriever-stack.js:40
  function wrapCallSite(unwrappedFrame) {
                       ^

RangeError: Maximum call stack size exceeded
(Use `node --trace-uncaught ...` to show where the exception was thrown)

Node.js v20.9.0
```
When at release-3.0 branch did checkout previous commit https://github.com/meteor/meteor/commit/09076cb3177016a843364939c1b2ab08c2190215 , building works without errors. After that "Update source-map-support" building does not work.

So to build for Linux arm64 at https://github.com/meteor/meteor/pull/12160#issuecomment-1846044210 , I had to add this additional 2nd step:
```
git checkout remotes/origin/release-3.0

git checkout 09076cb3177016a843364939c1b2ab08c2190215

git merge remotes/origin/feature-linux-arm
```

**Comment by Grubba27** [comment]
Hey @xet7 we are investigating this issue!

File: packages/minimongo/local_collection.js
Comment (MarcosSpessatto) [inline packages/minimongo/local_collection.js:662]: Aren't we forgetting to add the `upsert` scenario here? We are adding for the `updateAsync` but not for the "regular" one.

File: packages/meteor/url_common.js
Comment (MarcosSpessatto) [inline packages/meteor/url_common.js:47]: Any reason to replace localhost by default if the URL contains the localhost regex, instead of respecting the flag? Wouldn't it be a `&&`?

Comment (Grubba27) [reply packages/meteor/url_common.js:47]: This API had to be changed because of how node server handles fetching files inside of it. We are planning on documenting what changed (this will be probably a breaking change)

Comment (Grubba27) [reply packages/meteor/url_common.js:47]: We updated the tests for this function to match what is the behavior now. Only thing missing is docs

**Comment by harryadel** [comment]
Hi, I'm trying to update the [test application](https://github.com/Meteor-Community-Packages/meteor-collection2) of collection2 but I get this error. Is this a local issue of mine?

```bash
❯ meteor update --release 3.0-alpha.19
AggregateError                                
    at internalConnectMultiple (node:net:1114:18)
    at internalConnectMultiple (node:net:1177:5)
    at Timeout.internalConnectMultipleTimeout (node:net:1687:3)
    at listOnTimeout (node:internal/timers:575:11)
    at processTimers (node:internal/timers:514:7) {
  code: 'ETIMEDOUT',
  [errors]: [
    Error: connect ETIMEDOUT 188.114.97.7:443
        at createConnectionError (node:net:1634:14)
        at Timeout.internalConnectMultipleTimeout (node:net:1685:38)
        at listOnTimeout (node:internal/timers:575:11)
        at processTimers (node:internal/timers:514:7) {
      errno: -110,
      code: 'ETIMEDOUT',
      syscall: 'connect',
      address: '188.114.97.7',
      port: 443
    },
    Error: connect ENETUNREACH 2a06:98c1:3121::7:443 - Local (:::0)
        at internalConnectMultiple (node:net:1176:40)
        at Timeout.internalConnectMultipleTimeout (node:net:1687:3)
        at listOnTimeout (node:internal/timers:575:11)
        at processTimers (node:internal/timers:514:7) {
      errno: -101,
      code: 'ENETUNREACH',
      syscall: 'connect',
      address: '2a06:98c1:3121::7',
      port: 443
    },
    Error: connect ETIMEDOUT 188.114.96.7:443
        at createConnectionError (node:net:1634:14)
        at Timeout.internalConnectMultipleTimeout (node:net:1685:38)
        at listOnTimeout (node:internal/timers:575:11)
        at processTimers (node:internal/timers:514:7) {
      errno: -110,
      code: 'ETIMEDOUT',
      syscall: 'connect',
      address: '188.114.96.7',
      port: 443
    },
    Error: connect ENETUNREACH 2a06:98c1:3120::7:443 - Local (:::0)
        at internalConnectMultiple (node:net:1176:40)
        at Timeout.internalConnectMultipleTimeout (node:net:1687:3)
        at listOnTimeout (node:internal/timers:575:11)
        at processTimers (node:internal/timers:514:7) {
      errno: -101,
      code: 'ENETUNREACH',
      syscall: 'connect',
      address: '2a06:98c1:3120::7',
      port: 443
    }
  ]
}

```

**Comment by Grubba27** [comment]
##  Meteor v3.0-alpha.20 is out!
You can test it by running:

```bash
meteor update --release 3.0-alpha.20
```

In this release, we have a few great additions:

- Call async is now working correctly [pr](https://github.com/meteor/meteor/pull/12907)

### How is working now

For testing the changes, you can check this snippet:

```js
// Method created on the client
Meteor.methods({
  getData() {
    return "not data for the client";
  }
})

// Method created on the server
Meteor.methods({
  getData() {
    return LinksCollection.find().fetch();
  }
})

// Calling method on the client
const promise = Meteor.callAsync('getData');
console.log('getData SERVER', await promise); // Logs data returned from the server
console.log('getData CLIENT', await promise.stubPromise); //  Logs "not data for the client"
```

- moved compileForShell from ecmascript [pr](https://github.com/meteor/meteor/pull/12927)
- All changes that came in 2.14 are now in 3.0 [pr](https://github.com/meteor/meteor/pull/12795)

File: packages/meteor/url_common.js
Comment (tassoevan) [reply packages/meteor/url_common.js:47]: - If `url` **matches** against `locationRegex` and `options.replaceLocalhost` is **true**, this line will perform a string replacement in `url`.
- If `url` **matches** against `locationRegex` and `options.replaceLocalhost` is **false**, this line will perform a string replacement in `url`.
- If `url` **doesn't match** against `locationRegex` and `options.replaceLocalhost` is **true**, this line will do a trivial assignment (`url = url;`) because `url.replace()` will not find a matching substring; it could be just skipped.
- If `url` **doesn't match** against `locationRegex` and `options.replaceLocalhost` is **false**, this line will be skipped.

The `if` condition is unnecessary and just adds a regexp lookup (`localhostRegex.test(url)`) that `String.prototype.replace` already does.

**Comment by harryadel** [comment]
Given the recent merges, are you guys to release a new beta?

**Comment by harryadel** [comment]
:tada:

**Comment by xet7** [comment]
@Grubba27 @StorytellerCZ 

Any idea why I had to revert from Meteor 2.15 to 2.14 to fix some bugs at https://github.com/wekan/wekan/commit/b5f4be36d4577c391ad19c5e5dd9de3036b8998f ?

I don't know is it related to https://github.com/wekan/wekan/pull/5253 in any way.

**Comment by StorytellerCZ** [comment]
@xet7 potentially from what I have seen. Either way this is not related to this PR.

**Comment by filipenevola** [comment]
Hey, @ Grubba27, I don't think beta 4 is available yet. Is that correct? 

I see many commits, but no versions are being announced.

Let me know if I could help.

**Comment by Grubba27** [comment]
Hello, guys. We have great news: Meteor.js v3.0-beta.4 is out! and it comes with the following updates:

# New Features:

- ARM Support: You can run Meteor 3.0 on Raspberry Pi, AWS Graviton, and all ARM-based Linux machines, creating new opportunities for Meteor projects.
- MongoDB 7 (Ported from [2.15](https://blog.meteor.com/new-meteor-js-2-15-and-mongodb-7-f358bc304250)); 


## Bug fixes:
- [The Async iterator now works](https://github.com/meteor/meteor/pull/12989) (`for await (const item of Collection.find())`);
- [OAuth is now working as it should;](https://github.com/meteor/meteor/pull/13018)
- [Meteor.userAsync now works in publications;](https://github.com/meteor/meteor/pull/13019)
- [Isomorphism in Collection is back;](https://github.com/meteor/meteor/pull/12969)

### Known bugs:

Recently created react projects come with a bug regarding react-meteor-data. To solve it: `meteor remove react-meteor-data` and then `meteor add react-meteor-data@3.0.0-beta300.1`


### Blaze:
-  [v2.8.0 is out!](https://github.com/meteor/blaze/pull/446)  with added support for Promises in #if and #each
- The full changelog is [here](https://github.com/meteor/blaze/releases/tag/v2.8.0).


## How to test it?

### Existing projects:

```sh
meteor update --release 3.0-beta.4
```

### Starting a project:

```sh
meteor create app-name --release 3.0-beta.4
```

Screenshot 2024-02-22 at 16 10 17

**Comment by simonwebs** [comment]
Great work done by the @meteor team. Please it easy and simple to configure ssr. Or if possible throw some light on that.

**Comment by simonwebs** [comment]
The Isomorphic I hear been part with v3 is the one killing me. Thanks for this.

**Comment by TylerThompson** [comment]
Are there any current plans to natively support svelte in meteor?

**Comment by Grubba27** [comment]
We have a new release:  `meteor 3.0.0-beta.6`

## Updates: 

- Node was updated to v20.11.1
- Npm was updated to v10.2.4
- Push to deploy works with meteor v3.0 apps.
The push to deploy docs is [here](https://cloud-guide.meteor.com/deploy-to-galaxy), and those who want to use the CLI to deploy your meteor v3 app can use the following command:

```sh
meteor deploy yourappname.meteorapp.com ––free ––mongo 
```

## Bug fixes:
- [tests: Meteor.userAsync not ready in tests](https://github.com/meteor/meteor/pull/13027)
- [Removing broken methods in version 3](https://github.com/meteor/meteor/pull/13030)
- [Finished springboard migration to async](https://github.com/meteor/meteor/pull/13032)
- [Fixed linker bug that needed a workaround](https://github.com/meteor/meteor/pull/13031)

## Known bugs:

Projects that use React can get issues with `react-meteor-data`. Make sure that you are using the `3.0.0-beta.1` version. Check your `.meteor/versions file`. 
If your app is not using the correct version, you can  solve it by running: `meteor remove react-meteor-data`, and then `meteor add react-meteor-data@3.0.0-beta300.1`

## How to test it?

### Existing projects:

```sh
meteor update --release 3.0-beta.6
```
### Starting a project:

```sh
meteor create app-name --release 3.0-beta.6
```

Screenshot 2024-02-28 at 13 56 03

**Comment by xet7** [comment]
@Grubba27 

Installing Meteor with `npx meteor@beta` show some message, that installing at Linux arm64 is not supported?

**Comment by nachocodoner** [comment]
> @Grubba27
> 
> Installing Meteor with `npx meteor@beta` show some message, that installing at Linux arm64 is not supported?

Which is the exact message you get? I used this command on my M1 machine and it worked well.

**Comment by Grubba27** [comment]
@xet7 now that you have mentioned, that is an issue here on this code:
https://github.com/meteor/meteor/blob/b5975fcd9f24666b0a0ca3d90f9ddb2f47526d80/npm-packages/meteor-installer/install.js#L65-L84

Especially this line, if it is arm Linux it uses the `aarch64`

https://github.com/meteor/meteor/blob/b5975fcd9f24666b0a0ca3d90f9ddb2f47526d80/npm-packages/meteor-installer/install.js#L88

We should update the installer to get arm Linux bundle @nachocodoner 

Also, I just checked, and the meteor has not built a bundle for Arm Linux since beta 4. I've updated it. Now, we will have a dev bundle for every release for every arch.

**Comment by xet7** [comment]
@Grubba27 

At M1 Air, using Asahi Fedora Linux:
```
wekan@air:~$ uname -a
Linux air 6.6.3-414.asahi.fc39.aarch64+16k #1 SMP PREEMPT_DYNAMIC Sun Mar 24 19:44:17 UTC 2024 aarch64 GNU/Linux
wekan@air:~$ npx meteor@beta
The current architecture is not supported in this version:  arm64 . Try Meteor 2.5.1-beta.3 or above.
```

**Comment by xet7** [comment]
@Grubba27 

```
wekan@air:~$ node -v
v20.12.0
wekan@air:~$ npm -v
10.5.0
```

**Comment by xet7** [comment]
@Grubba27 

Now also building from branch is broken. Can not use Linux arm64 at all, at M1 Air.
Is there any other way to build with Meteor at Linux arm64 ?
```
git clone https://github.com/meteor/meteor .meteor

cd .meteor

git checkout remotes/origin/release-3.0

./scripts/generate-dev-bundle.sh

CHECKOUT DIR IS /home/wekan/.meteor
BUILDING DEV BUNDLE 20.11.1.2 IN /tmp/generate-dev-bundle-75zmk19i
Downloading Node from https://nodejs.org/dist/v20.11.1/node-v20.11.1-linux-arm64.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   155  100   155    0     0    644      0 --:--:-- --:--:-- --:--:--   645

gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now
Downloading Node from https://nodejs.org/download/rc/v20.11.1/node-v20.11.1-linux-arm64.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   146    0   146    0     0    435      0 --:--:-- --:--:-- --:--:--   434

gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now
```

**Comment by xet7** [comment]
@Grubba27

Now building from `release-3.0` branch works at Linux arm64 ! Thanks for fixing !

Command `npx meteor@beta` does not work yet.

**Comment by Grubba27** [comment]
> Command npx meteor@beta does not work yet.

hey @xet7! we are working on that, we will update once we have an status on this task.

**Comment by nachocodoner** [comment]
@xet7: We have already provided the fix on the installer and the support of Linux ARM. Could you test with the latest published version? `npx meteor@beta`

I even tested it on a Linux ARM VM on my Mac M1. More details [here](https://github.com/meteor/meteor/pull/13077)

**Comment by xet7** [comment]
@nachocodoner 

Thanks! Now `npx meteor@beta` works at Asahi Fedora at M1 Air !

File: .github/workflows/npm-meteor-promise.yml
Comment (StorytellerCZ) [inline .github/workflows/npm-meteor-promise.yml:21]: If this is being reversed for Fibers, then 14.x is good enough. No need to run another test for 12.x.

**Comment by xet7** [comment]
Installing rc does not work at Ubuntu Linux arm64 running at M1 Air:
```
npx meteor@rc
=> Arch: os.linux.aarch64
=> Meteor Release: 3.0-rc.1
Downloading |░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░| 0%node:events:497
      throw er; // Unhandled 'error' event
      ^

Error: Response status was 403
    at ClientRequest.<anonymous> (/home/ubuntu/.npm/_npx/83759a2e895bdfec/node_modules/node-downloader)
    at Object.onceWrapper (node:events:634:26)
    at ClientRequest.emit (node:events:519:28)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (node:_http_client:698:27)
    at HTTPParser.parserOnHeadersComplete (node:_http_common:119:17)
    at TLSSocket.socketOnData (node:_http_client:540:22)
    at TLSSocket.emit (node:events:519:28)
    at addChunk (node:internal/streams/readable:559:12)
    at readableAddChunkPushByteMode (node:internal/streams/readable:510:3)
    at Readable.push (node:internal/streams/readable:390:5)
Emitted 'error' event on b instance at:
    at ClientRequest.<anonymous> (/home/ubuntu/.npm/_npx/83759a2e895bdfec/node_modules/node-downloader)
    at Object.onceWrapper (node:events:634:26)
    [... lines matching original stack trace ...]
    at Readable.push (node:internal/streams/readable:390:5) {
  status: 403,
  body: ''
}

Node.js v20.13.1
```

**Comment by xet7** [comment]
Same Linux arm64 problem at OrangePi 5 running Ubuntu 24.04 arm64:

meteor-rc-orangepi

https://joshua-riek.github.io/ubuntu-rockchip-download/

https://github.com/Joshua-Riek/ubuntu-rockchip/releases

https://github.com/Joshua-Riek/ubuntu-rockchip/wiki/Ubuntu-24.04-LTS

http://www.orangepi.org/html/hardWare/computerAndMicrocontrollers/service-and-support/Orange-pi-5.html

**Comment by xet7** [comment]
Now installing `npx meteor@rc` works at Linux arm64. Thanks for the fix !

**Pull request merged**
